testing/deprecated_test.py

```diff
@@ -178,3 +178,11 @@ def test_hookproxy_warnings_for_fspath(tmp_path, hooktype, request):
     assert l1 < record.lineno < l2
 
     hooks.pytest_ignore_collect(config=request.config, fspath=tmp_path)
+
+
+def test_warns_none_is_deprecated():
+    with pytest.warns(
+        PytestDeprecationWarning,
+        match="Please pass an explicit Warning type or tuple of Warning types.",
+    ):
+        pytest.warns(None)
```

Comment:
something in me would like to see this as a `with ,,,: pass`

but thats personal taste ^^

Reply:
Could you elaborate? I didn't get that :D

Reply:
Ronny means writing:

```python
        PytestDeprecationWarning,
        match="Please pass an explicit Warning type or tuple of Warning types.",
    ):
        with pytest.warns(None):
            pass
```

so that it's obviously exercising the context manager.  We happen to know that this runs exactly the same code, but adding the `with` means that it's obvious to any later readers that this test applies to usage as part of a context manager.